src/scan.ts

```diff
@@ -18,16 +18,18 @@
  * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
  */
 
-export type ScanOptions = {
-  serverUrl: string;
-  token: string;
-  jvmOptions: string[];
-  options?: { [key: string]: string };
-  caPath: string;
-  logLevel?: string;
-  verbose?: boolean;
-};
+import { log, LogLevel } from './logging';
+import { getPlatformInfo } from './platform';
+import { ScanOptions } from './types';
 
 export async function scan(scanOptions: ScanOptions, cliArgs?: string[]) {
-  // TODO: NPMSCAN-2 new bootstrapper sequence
+  log(LogLevel.DEBUG, 'Finding platform info');
+  const platformInfo = getPlatformInfo();
+  log(LogLevel.INFO, 'Platform: ', platformInfo);
```

Comment:
I need to conditionally skip this with the presence of `sonar.scanner.skipArchDetection` 

@7PH let me know if you have any process or suggestion on how we want to do this 🙇🏻

Reply:
After [this PR](https://github.com/SonarSource/sonar-scanner-npm/pull/111) you'll get the `properties` object available right before these lines, so we'll be able to use the object to decide whether to detect or not